### DISCUSSION TITLE
Extending the Default Menu and Tooltip

### DISCUSSION THREAD

**Discussion by tohhongxiang** [Milkdown/milkdown#615] [General]
Hi, I was trying to include a highlight plugin for milkdown. But currently I'm running into the problem of including a button for it within the menu

I'm currently using `@milkdown/plugin-menu` to create the menu, and here's what I have:

```js
// ...

.use(
    menu.configure(menuPlugin, {
                config: [ 
                    ...defaultConfig,
                    [
                        {
                            type: 'button',
                            icon: 'Highlight' as any,
                            key: ToggleHighlightedText,
                        }
                    ]
                ]
            })
)

// ....
```

The current problem is that `icon` only accepts a few specific strings (such as "h1" or "delete") which maps to material-icon menus. I would like to include my own custom icon instead. How would I go about doing that?

The next question I have as well is regarding the tooltip. I would like the user to be able to select the text and then highlight it (similar to how you can select a text and bold it). But currently, theres not much documentation about how to include additional actions in the tooltip. I would like some general directions regarding how to approach this problem.

Thanks! Love this library so far. It checks the most boxes for me (especially with the math plugin)

**Comment by Saul-Mirone** [comment]
I'll write you a example later.

**Comment by Saul-Mirone** [comment]
I find that it's not possible to provide custom menu icon for now. I plan to support it in next version. Will come back to you once I finished.
